src/echo/web/logger.gleam

```diff
@@ -22,7 +22,8 @@ fn format_log_line(request: Request(a), response: Response(b)) -> String {
 pub fn middleware(service: Service(a, b)) -> Service(a, b) {
   fn(request) {
     let response = service(request)
-    io.println(format_log_line(request, response))
+    format_log_line(request, response)
+    |> io.println
```

Comment:
_leaving a comment just for the record_

An alternative implementation that has more functional style using [`tap` function](https://hexdocs.pm/gleam_stdlib/gleam/function.html#tap)

```gleam
import gleam/function

// ...

pub fn middleware(service: Service(a, b)) -> Service(a, b) {
  fn(request) {
    service(request)
    |> function.tap(fn(response) {
      response
      |> format_log_line(request, _)
      |> io.println
    })
  }
}
```